moveit_ros/moveit_servo/src/servo_node.cpp

```diff
@@ -106,6 +108,50 @@ ServoNode::ServoNode(const rclcpp::NodeOptions& options)
   servo_ = std::make_unique<moveit_servo::Servo>(node_, planning_scene_monitor_, std::move(param_listener));
 }
 
+void ServoNode::validateParams(const servo::Params servo_params)
+{
+  bool has_error = false;
+  if (servo_params.hard_stop_singularity_threshold <= servo_params.lower_singularity_threshold)
+  {
+    RCLCPP_WARN(LOGGER, "Parameter 'hard_stop_singularity_threshold' "
+                        "should be greater than 'lower_singularity_threshold.' "
+                        "Check the parameters YAML file used to launch this node.");
+    has_error = true;
+  }
+
+  if (!servo_params.publish_joint_positions && !servo_params.publish_joint_velocities &&
+      !servo_params.publish_joint_accelerations)
+  {
+    RCLCPP_WARN(LOGGER, "At least one of publish_joint_positions / "
+                        "publish_joint_velocities / "
+                        "publish_joint_accelerations must be true. "
+                        "Check the parameters YAML file used to launch this node.");
+    has_error = true;
+  }
+
+  if ((servo_params.command_out_type == "std_msgs/Float64MultiArray") && servo_params.publish_joint_positions &&
+      servo_params.publish_joint_velocities)
+  {
+    RCLCPP_WARN(LOGGER, "When publishing a std_msgs/Float64MultiArray, "
+                        "you must select positions OR velocities."
+                        "Check the parameters YAML file used to launch this node.");
+    has_error = true;
+  }
+
+  if (servo_params.scene_collision_proximity_threshold < servo_params.self_collision_proximity_threshold)
+  {
+    RCLCPP_WARN(LOGGER, "Parameter 'self_collision_proximity_threshold' should probably be less "
+                        "than or equal to 'scene_collision_proximity_threshold'."
+                        "Check the parameters YAML file used to launch this node.");
+    has_error = true;
+  }
+
+  if (has_error)
+  {
+    throw std::runtime_error("Failed to initialize servo");
```

Comment:
If this throws, then the logger messages above should either be `RCLCPP_ERROR` or `RCLCPP_FATAL`, not warnings, right?

Reply:
Not sure about that, it was like this in the old implementation.
So I followed that pattern.

Reply:
Should the failed checks be considered as errors ?
or can we have a single error message at the end before throwing?
Stating servo failed to initialize due to invalid parameter values

Reply:
You can definitely do that. Start with an empty `std::string` and append any of the logger messages to it, then if you throw an error you tack that string on there.

Reply:
Heads-up, clang-tidy may yell at you if you use the `+` operator for this. So do:

```
std::string error_text = "";

if (some_bad_thing_happened)
{
    my_error = "Everything is broken.";
    RCLCPP_ERROR(LOGGER, my_error.c_str());
    my_error.append(error_text).append("\n");
    has_error = true;
}

if (has_error)
{
    throw std::runtime_error("Failed with reason: ".append(error_text));
}
```

Reply:
Feels like this would make it less readable.
Since multiple checks could fail and the error message will get quite big (huge multi-line error message)
I think we can convert all the warnings to errors instead and just throw with a `Servo failed to initialize : Invalid parameters` message.

Reply:
That's fine too so long as the errors are being logged